src/components/WalletModal/index.tsx

```diff
@@ -174,10 +230,21 @@ export default function WalletModal({
     if (account && account !== lastActiveWalletAddress) {
       const walletType = getConnectionName(getConnection(connector).type, getIsMetaMask())
       sendAnalyticsEventAndUserInfo(account, walletType, chainId)
+      setLogWalletBalances(true)
     }
     setLastActiveWalletAddress(account)
   }, [lastActiveWalletAddress, account, connector, chainId])
 
+  // Send wallet balance info once it becomes available.
+  useEffect(() => {
+    if (!tokenBalancesIsLoading && logWalletBalances && balances && nativeCurrencyBalanceUsdValue) {
+      const nativeCurrencyBalanceUsd =
+        native && nativeCurrencyBalanceUsdValue ? parseFloat(nativeCurrencyBalanceUsdValue) : 0
+      sendAnalyticsWalletBalanceUserInfo(balances, nativeCurrencyBalanceUsd)
+      setLogWalletBalances(false)
```

Comment:
might be better to name this `setWalletBalancesLogged` and set to true? otherwise it's unclear what the value si for

Reply:
sounds good done